core/types/tx_blob.go

```diff
@@ -20,28 +20,48 @@ import (
 	"math/big"
 
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto/kzg4844"
 	"github.com/ethereum/go-ethereum/params"
-	"github.com/holiman/uint256"
 )
 
 // BlobTx represents an EIP-4844 transaction.
 type BlobTx struct {
-	ChainID    *uint256.Int
+	ChainID    *big.Int
 	Nonce      uint64
-	GasTipCap  *uint256.Int // a.k.a. maxPriorityFeePerGas
-	GasFeeCap  *uint256.Int // a.k.a. maxFeePerGas
+	GasTipCap  *big.Int // a.k.a. maxPriorityFeePerGas
+	GasFeeCap  *big.Int // a.k.a. maxFeePerGas
 	Gas        uint64
 	To         common.Address
-	Value      *uint256.Int
+	Value      *big.Int
 	Data       []byte
 	AccessList AccessList
-	BlobFeeCap *uint256.Int // a.k.a. maxFeePerDataGas
+	BlobFeeCap *big.Int // a.k.a. maxFeePerDataGas
 	BlobHashes []common.Hash
 
 	// Signature values
-	V *uint256.Int `json:"v" gencodec:"required"`
-	R *uint256.Int `json:"r" gencodec:"required"`
-	S *uint256.Int `json:"s" gencodec:"required"`
+	V *big.Int `json:"v" gencodec:"required"`
+	R *big.Int `json:"r" gencodec:"required"`
+	S *big.Int `json:"s" gencodec:"required"`
+}
+
+type BlobTxWithBlobs struct {
```

Comment:
This should not exist in the consensus types at all. This is a network level thing, pls move it to eth/protocols/eth